test/python/synthesis/test_multi_controlled_synthesis.py

```diff
@@ -192,6 +193,14 @@ def test_mcx_noaux_v24(self, num_ctrl_qubits: int):
             XGate(), num_ctrl_qubits, synthesized_circuit, clean_ancillas=False
         )
 
+    @data(1, 2, 3, 4, 5, 6, 7, 8)
+    def test_mcx_noaux_hp24(self, num_ctrl_qubits: int):
```

Comment:
I think that some more tests are needed.

- Since there are two different synthesis methods, for small and large number of qubits, should we also check the large here? maybe using the `Statevector` simulator?

- Can we add a  `test_mcx_noaux_hp24_cx_count` test? based on some formulas from the paper (Theorem 4.5)? 
usually we check it for `n=5,10,15`, should we also check `n>22` here?

- The bound in the tests in line 469 should be updates:

```
if isinstance(base_gate, (XGate, YGate, ZGate, HGate)):
            # MCX gate and other locally equivalent multi-controlled gates
            expected = {1: 1, 2: 6, 3: 14, 4: 36, 5: 84, 6: 140, 7: 220, 8: 324}
```

Reply:
I have addressed the second and the third comments (both are great suggestions).

Regarding the first point. I completely agree that it would be very nice to add some correctness checking for large values of ``n``, especially that the algorithm follows different compilation schemes for large values of ``n``. However, at this point constructing either ``Operator`` or ``Statevector`` is impossible, do you agree? What can we do instead?